docs/react-v9/contributing/rfcs/react-components/convergence/extended-design-tokens.md

```diff
@@ -0,0 +1,455 @@
+  ColorTokens &
+  ButtonTokens; // Addition
+```
+
+### Other options considered
+
+| Option                                                          | Pros                                                                                                                                     | Cons                                                                                                                                                                                                                                               |
+| --------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **Option A: Add all tokens to current FluentProvider**          | • Simple<br>• Builds on existing API<br>• No additional deps for partners                                                                | • Theme typings will get large as we'd include all component tokens<br>• Tokens object will get larger but not massive since we'd only ship values for component groups and not component tokens (these are optional slots)                        |
+| Option B.1: Create separate semantic FluentProvider             | • Separates Fluent Semantic Tokens (FSTs) from FluentProvider and Theme.<br>• Optional layering so partners only pull it in when needed. | • Requires extra layer<br>• May not be as intuitive as just using the existing provider already in applications<br>• May not provide much value since the components still need to have some level of awareness of the group and component tokens. |
+| Option B.2: Create separate semantic FluentProvider and package | • Same pros as above but with added flexibility around packaging and versioning                                                          | See above                                                                                                                                                                                                                                          |
+
+## Update components to optionally consume expanded tokens
+
+Once exposed in the theme, the new tokens could be accessed in existing components simply by updating existing references to alias tokens with var(), passing the more precise token as the first argument and retaining original alias token as the second.
+
+For example, instead of:
+
+```javascript
+const useRootBaseClassName = makeStyles({ backgroundColor: 'var(--colorNeutralForeground1)' });
+```
+
+We would use this:
+
+```javascript
+const useRootBaseClassName = makeStyles({
+  backgroundColor: 'var(--buttonBackgroundColorRest, var(--colorNeutralForeground1))',
+});
+```
```

Comment:
- The same question as above about typos
- How are going to ensure that newly written rules are using tokens i.e. we transformed components, next day I will write a new style. What will stop me from misusing new tokens or completely skipping them?
- What CSS properties can be tokens? Could be `padding` a token? `display`, `boxSizing`? How to do we define this?

Reply:
I think the governance question is a good one but it isn't really a new problem introduced here. We have the same problem here. We could potentially look at a link or governance rule to help with control tokens? There's probably a tool we could also write that we pull tokens used on the design end into a check here. But those tools don't exist yet and I haven't thought enough about it to realize all the edge cases or draw backs. I think it's a good point though and definitely an issue we should look to solve.

In terms of what should be a token, if we're thinking about design tokens themselves it should be how we codify the design system itself. We tokenize things where we've made explicit design choices that exemplify Fluent. It's a bit vague (likely intentionally) but that's really what distinguishes tokens from just variables. IMHO, we shouldn't think about tokens and variables as the same thing. There's a lot more intentionality behind tokens than just a shared value.